contracts/SafeMath.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity ^0.4.21;
+
+/**
+ * @title SafeMath
+ * @dev Math operations with safety checks that throw on error
+ */
+library SafeMath {
+
+  /**
+  * @dev Multiplies two numbers, throws on overflow.
+  */
+  function mul(uint256 a, uint256 b) internal pure returns (uint256) {
+    if (a == 0) {
```

Comment:
Should the logic be if a == 0 OR b == 0 ?

Reply:
SafeMath.sol is from https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/contracts/math/SafeMath.sol

They kind of know what they are doing, at least much better than I do.

Regarding your comment, no, for the check that follows it's only vital that a is not 0.

Reply:
Well, if either A or B is 0, the function will return 0.

Reply:
Yes, but if a is 0 the function would in fact throw an division by zero exception. In contrast to b = 0.

Reply:
I guess that's true.